integrations/access/accessrequest/app.go

```diff
@@ -345,6 +415,14 @@ func (a *App) getMessageRecipients(ctx context.Context, req types.AccessRequest)
 	// This can happen if this set contains the channel `C` and the email for channel `C`.
 	recipientSet := common.NewRecipientSet()
 
+	recipients := a.recipientsFromAccessMonitoringRules(ctx, req)
+	for _, recipient := range recipients.ToSlice() {
+		recipientSet.Add(recipient)
+	}
+	if len(recipientSet.ToSlice()) != 0 {
+		return recipientSet.ToSlice()
```

Comment:
```suggestion
	if s := recipientSet.ToSlice(); len(s) != 0 {
		return s
```

This way you're not allocating two separate slices to hold the same data and throwing one of them away.